homeassistant/components/sms/config_flow.py

```diff
@@ -21,13 +34,17 @@ async def get_imei_from_config(hass: core.HomeAssistant, data):
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
     device = data[CONF_DEVICE]
-    config = {"Device": device, "Connection": "at"}
+    connection_mode = "at"
+    baud_speed = str(data.get(CONF_BAUD_SPEED, DEFAULT_BAUD_SPEED))
+    if baud_speed != str(DEFAULT_BAUD_SPEED):
```

Comment:
`DEFAULT_BAUD_SPEED` is already a string?

```suggestion
    if baud_speed != DEFAULT_BAUD_SPEED:
```